src/Infrastructure/Adapter/ApiOrganizationFetcher.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Infrastructure\Adapter;
+
+use App\Application\ApiOrganizationFetcherInterface;
+use App\Domain\User\Exception\OrganizationNotFoundException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ApiOrganizationFetcher implements ApiOrganizationFetcherInterface
+{
+    public function __construct(
+        private HttpClientInterface $organizationFetcherClient,
+    ) {
+    }
+
+    public function findBySiret(string $siret): array
+    {
+        $response = $this->organizationFetcherClient->request(
+            'GET',
+            'search',
+            [
+                'query' => [
+                    'q' => $siret,
+                    'est_collectivite_territoriale' => 'true',
+                ],
+            ],
+        );
+
+        $data = $response->toArray();
+
+        if (0 === $data['total_results']) {
+            throw new OrganizationNotFoundException($siret);
+        }
+
+        return [
+            'name' => $data['results'][0]['nom_complet'],
+        ];
```

Comment:
(Détail) Vu qu'il n'y a qu'un seul champ dans l'array, est-ce qu'il ne faudrait pas mieux transformer ça en `findNameBySiret(): string` ?

```suggestion
        return $data['results'][0]['nom_complet'];
```

Reply:
Aujourd'hui nous n'utilisons que le name, je ne sais pas si demain nous aurons besoin d'avoir d'autres informations, c'est pour ça que je l'ai laissé générique.